programs/locker/src/instructions/cancel_vesting_escrow.rs

```diff
@@ -0,0 +1,132 @@
+use anchor_spl::token::{Token, TokenAccount};
+use solana_program::program::invoke_signed;
+use spl_token::instruction::close_account;
+
+use crate::*;
+use crate::util::token::transfer_to_recipient;
+
+/// Accounts for [locker::cancel_vesting_escrow].
+#[derive(Accounts)]
+#[event_cpi]
+pub struct CancelVestingEscrow<'info> {
+    /// Escrow.
+    #[account(
+        mut,
+        has_one = creator
+    )]
+    pub escrow: AccountLoader<'info, VestingEscrow>,
+
+    #[account(
+        mut,
+        associated_token::mint = recipient_token.mint,
+        associated_token::authority = escrow
+    )]
+    pub escrow_token: Box<Account<'info, TokenAccount>>,
+
+    #[account(mut)]
+    pub creator_token: Box<Account<'info, TokenAccount>>,
+
+    #[account(mut)]
+    pub recipient_token: Box<Account<'info, TokenAccount>>,
+
+    /// CHECKED: the creator will receive the rent back
+    #[account(mut)]
+    pub creator: UncheckedAccount<'info>,
+
+    /// Signer.
+    #[account(mut)]
+    pub signer: Signer<'info>,
+
+    /// Token program.
+    pub token_program: Program<'info, Token>,
+
+    /// System program.
+    pub system_program: Program<'info, System>,
+}
+
+impl<'info> CancelVestingEscrow<'info> {
+    fn close_escrow_token(&self) -> Result<()> {
+        let escrow = self.escrow.load()?;
+        let escrow_seeds = escrow_seeds!(escrow);
+
+        invoke_signed(
+            &close_account(
+                self.token_program.key,
+                self.escrow_token.to_account_info().key,
+                self.creator.key,
+                &self.escrow.key(),
+                &[],
+            )?,
+            &[
+                self.token_program.to_account_info(),
+                self.escrow_token.to_account_info(),
+                self.creator.to_account_info(),
+                self.escrow.to_account_info(),
+            ],
+            &[&escrow_seeds[..]],
+        )?;
+
+        Ok(())
+    }
+
+    fn validate(&self) -> Result<()> {
+        let escrow = self.escrow.load()?;
+        require!(
+            escrow.recipient == self.recipient_token.owner && escrow.token_mint == self.recipient_token.mint,
+            LockerError::InvalidRecipientTokenAccount,
+        );
+
+        require!(
+            escrow.creator == self.creator_token.owner && escrow.token_mint == self.creator_token.mint,
+            LockerError::InvalidCreatorTokenAccount,
+        );
+
+        Ok(())
+    }
+}
+
+pub fn handle_cancel_vesting_escrow(
+    ctx: Context<CancelVestingEscrow>
+) -> Result<()> {
+    ctx.accounts.validate()?;
+
+    let mut escrow = ctx.accounts.escrow.load_mut()?;
+    let signer = ctx.accounts.signer.key();
+    escrow.validate_cancel_actor(signer)?;
```

Comment:
pls match with coding style in update_escrow_recipient

Reply:
updated